### ISSUE TITLE
linker command failed with exit code 1

### ISSUE THREAD

**Issue by josiahallencomm** [FormidableLabs/react-native-app-auth#211]
This is a bug that has been addressed before, but none of the solutions are working for me.

I am trying to run `Product -> Archive` in Xcode so I can export an `.ipa` file to upload to HockeyApp. This process worked until I added `react-native-app-auth` due to a complaint from the Apple AppStore reviewers.

These are the dependencies from my `package.json` file:

```json
{
  "dependencies": {
    "react": "16.4.1",
    "react-native": "^0.56.0",
    "react-native-app-auth": "^4.0.0",
    "react-native-elements": "^0.19.1",
    "react-native-fs": "^2.12.0",
    "react-native-remote-svg": "^1.2.0",
    "react-native-svg": "^6.4.1",
    "react-native-svg-from-uri": "^1.0.2",
    "react-native-vector-icons": "^4.6.0",
    "react-native-zip-archive": "^3.0.1",
    "react-navigation": "^2.6.2",
    "rn-fetch-blob": "^0.10.13"
  },
  "devDependencies": {
    "babel-jest": "23.2.0",
    "babel-preset-react-native": "^5.0.2",
    "jest": "23.3.0",
    "react-test-renderer": "16.4.1"
  }
}
```

I am running Xcode 10.1 on macOS High Sierra 10.13.6.

I have followed the various steps in #118 such as running `react-native link` and opening/building `.xcworkspace` instead of `.xcodeproj` (which does allow me to navigate the project files), but I still see the same error.

I have followed the steps for #16 which I was directed to by #106, but I don't even have the option to add `libAppAuth.a` and am still getting the same error.

I looked at a few issues on Stackoverflow including [this one](https://stackoverflow.com/questions/48564028/react-native-build-release-linker-command-failed-with-exit-code-1) which suggests running `pod deintegrate` and then `pod install`, but that didn't fix the issue.

When I browse the pods from the workspace I see all of the files under `Pods/AppAuth`, but `Products/libAppAuth.a` is red because the file doesn't exist at `/Users/.../Library/Developer/Xcode/DerivedData/.../Build/Products/Release-iphoneos/AppAuth/libAppAuth.a`. It is the same case for the `Products/libPods-...App.a` file:

![screen shot 2018-11-21 at 12 39 03 pm](https://user-images.githubusercontent.com/15255769/48864792-e353f480-ed8a-11e8-89d7-0d86a0cde3ae.png)

The only place I can find `libAppAuth.a` is under my RN project directory at `ios/build/Build/Products/Debug-iphonesimulator/AppAuth/libAppAuth.a`. It **does not** exist under `ios/build/Build/Products/Release-iphonesimulator/...`.

I am not very familiar with Xcode or the whole process of building iOS apps, so I have quickly run out of ideas for troubleshooting. Any help is very appreciated!

**Comment by josiahallencomm** [comment]
It ended up being fixed by #16... I have been messing with this for hours to no avail, and then all of the sudden `libAppAuth.a` appeared as an option to add under `Linked Frameworks and Libraries` in the project's settings. There must have been some specific order of doing the various tasks that finally made it appear. After adding `libAppAuth.a` and removing `libPods-...App.a` the build succeeded!

**Comment by Mattchewone** [comment]
I just wanted to say that I ran into the same issue. I couldn't figure out how to do it with CocoaPods it for some reason just wouldn't create the `libAppAuth` module so I couldn't add it as a linked library.

I found that swapping `Carthage` worked and have managed to get a running app on to my physical device.
